library/common-library/templates/_privileged.tpl

```diff
@@ -0,0 +1,37 @@
+{{- /*
+This file contains helpers to handle whether the chart should assume it is fine to deploy itself with elevated
+privileges, thus changing the defaults for hostNetwork, securityContexts, etc.
+Please note that either local or global definitions of the aforementioned settings will override the privileged flag.
+*/ -}}
+
+{{- /*
+common.privileged is a helper that returns whether the chart should assume the user is fine deploying privileged pods.
+Chart writers should _not_ use this to populate securityContext.privileged directly, but rather to tweak their implementation of:
+- common.securityContext.containerDefaults
+- common.securityContext.podDefaults
+- common.hostNetwork.defaultOverride
+And then use the helpers this library provides to render those.
+*/ -}}
+{{- /* Chart writers should not rely on this helper directly. */ -}}
+{{- define "common.privileged" -}}
+{{- /* This allows us to use `$global` as an empty dict directly in case `Values.global` does not exists. */ -}}
+{{- $global := index .Values "global" | default dict -}}
+{{- /* `get` will return "" (empty string) if value is not found, and the value otherwise, so we can type-assert with kindIs */ -}}
+{{- if get .Values "privileged" | kindIs "bool" -}}
+    {{- with .Values.privileged -}}
+        {{- . -}}
+    {{- end -}}
```

Comment:
I think the `with` is cool while templating but it could somehow hide the things that it is doing.

This is the reason I chose in other PRs to use `if-else`. as we used in other helpers:
 - https://github.com/newrelic/helm-charts/blob/privileged/library/common-library/templates/_low-data-mode.tpl
 - https://github.com/newrelic/helm-charts/blob/privileged/library/common-library/templates/_staging.tpl
 - https://github.com/newrelic/helm-charts/blob/privileged/library/common-library/templates/_serviceaccount.tpl#L1-L38
 
There is no style guide yet but I would prefer to be the same as the other approved and merged. If the team prefers to switch to `with` instead of `if` I would change all at the same time after merging this properly to see al the changes of style at the same time.

```suggestion
    {{- if .Values.privileged -}}
        {{- .Values.privileged -}}
    {{- end -}}
```